vms/platformvm/txs/executor/proposal_tx_executor.go

```diff
@@ -644,164 +666,24 @@ func (e *ProposalTxExecutor) RewardValidatorTx(tx *txs.RewardValidatorTx) error
 		primaryNetworkValidator.StartTime,
 	)
 	if err != nil {
-		return fmt.Errorf("failed to calculate uptime: %w", err)
+		return false, fmt.Errorf("failed to calculate uptime: %w", err)
 	}
-
-	e.PrefersCommit = uptime >= expectedUptimePercentage
-	return nil
+	return uptime >= expectedUptimePercentage, nil
 }
 
-// GetNextStakerChangeTime returns the next time a staker will be either added
-// or removed to/from the current validator set.
-func GetNextStakerChangeTime(state state.Chain) (time.Time, error) {
-	currentStakerIterator, err := state.GetCurrentStakerIterator()
-	if err != nil {
-		return time.Time{}, err
-	}
-	defer currentStakerIterator.Release()
+func splitAmountByShares(totalAmount uint64, shares uint32, sharesAmountCap uint64) (uint64, uint64) {
```

Comment:
This should probably be exported - data platform needs to re-implement this currently for the P-chain parser. I think ideally I think we expose the first half of this function as an exported function from the `reward` package

Reply:
Exported and dropped the cap part. We won't need it until continuous staking restaking